drivers/flash/flash_page_layout.c

```diff
@@ -80,3 +80,31 @@ size_t flash_get_page_count(struct device *dev)
 
 	return count;
 }
+
+void flash_page_foreach(struct device *dev, flash_page_cb cb, void *data)
```

Comment:
need some code in Zephyr that uses this functions, preferably a testcase...

Reply:
I would drop this from the PR until you have something that uses it.

Reply:
I'm using it out of tree, in mcuboot. I'll add a test case.